src/js/outreach/views/opt-in_views.js

```diff
@@ -0,0 +1,153 @@
+      />
+    </div>
+    <div class="opt-in__field">
+      <label class="opt-in__field-label">Your date of birth</label>
+      <input
+        type="text"
+        class="input-primary opt-in__field-input js-dob"
+        placeholder="Enter your date of birth"
+        value="{{ dob }}"
+      />
+    </div>
+    <h3 class="opt-in__heading-text u-margin--t-32 u-margin--b-16">How may we share health resources with you?</h3>
+    <div class="opt-in__field">
+      <label class="opt-in__field-label">Your mobile phone number</label>
+      <input
+        type="text"
+        class="input-primary opt-in__field-input js-phone"
+        placeholder="Enter mobile phone number"
+        value="{{ phone }}"
+      />
+    </div>
+    <div class="opt-in__field">
+      <label class="opt-in__field-label">Your email address</label>
+      <input
+        type="text"
+        class="input-primary opt-in__field-input js-email"
+        placeholder="Enter email address"
+        value="{{ email }}"
+      />
+    </div>
+    <p class="opt-in__disclaimer">By clicking Submit you agree to receive SMS text message notifications. You may opt out at any time.</p>
+    <button class="opt-in__submit button--green w-100 js-submit" disabled>Submit</button>
+  `,
+  ui: {
+    firstName: '.js-first-name',
+    lastName: '.js-last-name',
+    dob: '.js-dob',
+    phone: '.js-phone',
+    email: '.js-email',
+    submit: '.js-submit',
+  },
+  triggers: {
+    'keyup @ui.firstName': 'change:first:name',
+    'keyup @ui.lastName': 'change:last:name',
+    'keyup @ui.dob': 'change:dob',
+    'keyup @ui.phone': 'change:phone',
+    'keyup @ui.email': 'change:email',
+    'click @ui.submit': 'click:submit',
+  },
+  onRender() {
+    this.setSubmitButtonState();
+  },
+  onChangeFirstName() {
+    this.model.set({ firstName: this.ui.firstName.val() });
+    this.setSubmitButtonState();
+  },
+  onChangeLastName() {
+    this.model.set({ lastName: this.ui.lastName.val() });
+    this.setSubmitButtonState();
+  },
+  onChangeDob() {
+    this.model.set({ dob: this.ui.dob.val() });
+    this.setSubmitButtonState();
+  },
+  onChangePhone() {
+    this.model.set({ phone: this.ui.phone.val() });
+    this.setSubmitButtonState();
+  },
+  onChangeEmail() {
+    this.model.set({ email: this.ui.email.val() });
+    this.setSubmitButtonState();
+  },
+  disableSubmitButton() {
+    this.ui.submit.prop('disabled', true);
+  },
+  enableSubmitButton() {
+    this.ui.submit.prop('disabled', false);
+  },
+  setSubmitButtonState() {
+    const firstName = this.ui.firstName.val();
+    const lastName = this.ui.lastName.val();
+    const dob = this.ui.dob.val();
+    const phone = this.ui.phone.val();
+    const email = this.ui.email.val();
+
+    if (!firstName || !lastName || !dob) {
+      this.disableSubmitButton();
+      return;
+    }
+
+    if (!phone && !email) {
+      this.disableSubmitButton();
+      return;
+    }
+
+    this.enableSubmitButton();
+  },
```

Comment:
Handles whether the submit button should have the `disabled` property or not.

The submit button is enabled if the following are true:

* The user has entered their first name, last name, and date of birth.
* The user has entered at least one communication method. Either email address or phone number.